### ISSUE TITLE
Reply behavior violates RFCs when a List-Post header is present

### ISSUE THREAD

**Issue by a-name-not-taken** [thunderbird/thunderbird-android#7305]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.603

### Where did you get the app from?

Google Play

### Android version

10

### Device model

SM-G960U1

### Steps to reproduce

Reply to a message that contains a "sender" header

### Expected behavior

Address the reply to the address in the "From" header or to the address in the "Reply-To" header if present.

### Actual behavior

The reply is addressed solely to the address in the "Sender" header.

For several years a simple "reply" to a message sent via a mailing list
went solely to the address in a "List-Post" header, if present, rather
than to the original author in the "From" or "Reply-to" headers. 
That seems to have finally been fixed, but now, as of V6.603, a simple
reply is being sent solely to the address in the "Sender" header, if
present. 

Neither behavior is or was correct at any time, and the problem is
exacerbated by the "To" field in the reply composition containing only
"+1", so the person replying has no idea that this unexpected and
unwanted behavior is present.

Could we PLEASE conform to the relevant RFCs? A reply is to be addressed
to the "From" address(es) in the original message unless a "Reply-To"
header is present, in which case the reply is to be addressed there. Full
stop. A Reply-all should go to the "From" or "Reply-To" as well as to the
"To", "CC", and, if present "Bcc" addresses placed in the same headers
going back. 

From RFC 5322 Section 3.6.2 (updated but unchanged by RFC 6854): 

"The "From:" field specifies the author(s) of the message, that is, the
mailbox(es) of the person(s) or system(s) responsible for the writing of
the message. The "Sender:" field specifies the mailbox of the agent
responsible for the actual transmission of the message. For example, if
a secretary were to send a message for another person, the mailbox of
the secretary would appear in the "Sender:" field and the mailbox of the
actual author would appear in the "From:" field." 

"The originator fields also provide the information required when
replying to a message. When the "Reply-To:" field is present, it
indicates the address(es) to which the author of the message suggests
that replies be sent. In the absence of the "Reply-To:" field, replies
SHOULD by default be sent to the mailbox(es) specified in the "From:"
field unless otherwise specified by the person composing the reply." 

And from Section 3.6.3: 

"When a message is a reply to another message, the mailboxes of the
authors of the original message (the mailboxes in the "From:" field) or
mailboxes specified in the "Reply-To:" field (if it exists) MAY appear
in the "To:" field of the reply since these would normally be the
primary recipients of the reply. If a reply is sent to a message that
has destination fields, it is often desirable to send a copy of the
reply to all of the recipients of the message, in addition to the
author. When such a reply is formed, addresses in the "To:" and "Cc:"
fields of the original message MAY appear in the "Cc:" field of the
reply, since these are normally secondary recipients of the reply. If a
"Bcc:" field is present in the original message, addresses in that field
MAY appear in the "Bcc:" field of the reply, but they SHOULD NOT appear
in the "To:" or "Cc:" fields."

**Comment by a-name-not-taken** [comment]
Here are the RFCs:

https://www.rfc-editor.org/rfc/rfc5322

https://www.rfc-editor.org/rfc/rfc6854

**Comment by cketti** [comment]
What makes you think K-9 Mail is using the `Sender` header when replying to a message? Can you post the message you are testing this with?

**Comment by a-name-not-taken** [comment]
> What makes you think K-9 Mail is using the `Sender` header when replying to a message? Can you post the message you are testing this with?

Thanks for following up. You're right.

It's possible I misread an X-Sender header for a Sender header, but I'm getting inconsistent results with the same K-9 version on mailing list messages from June, August, and October with nearly identical headers. I need to do a closer comparison among them to see what's actually going on.

For example, on the June message, a simple reply is going to a List-Post address, not the From address. There was no Reply-To address, and it did not go to the Sender address. In August and October, the reply went to the address in the identical From and Reply-To headers, even though the same List-Post header was present. Again, it did not go to the Sender address.

I thought the List-Post error was fixed, but it seems that it may actually be the addition of a Reply-To header that suppressed that. In any case, I need to scour those messages some more.

**Comment by a-name-not-taken** [comment]
Sorry for the confusion. The Sender header, although present, is not used as a reply address.
There IS a problem with "List-Post" headers being used and with too many addresses being used
in a reply-all.

What I see in messages I have received to which I have initiated a reply or reply-all is this:

For messages that include a "List-Post" header, a simple reply is addressed solely "To" the address in
the "List-Post" header UNLESS there is also a "Reply-To" header. In that case, it is addressed solely
"To" the address in the "Reply-To" header. This should instead be addressed to the original "From"
address unless there is a "Reply-To", in which case it should be addressed there. "List-Post" has
no role in replies.

A reply-all to a message containing a "List-Post" header is addressed "To" all of the addresses in
the "From", "To", and "List-Post" headers UNLESS there is also a "Reply-To" header. In that case,
it is addressed "To" all of the addresses in the "From", "To", and "Reply-To" headers. These
replies should instead be going "To" the "From" and "To" addresses OR to the "Reply-To" and "To"
addresses. "From" and "Reply-To" addresses should not both be used, nor should a "List-Post"
address. Addresses in the original "To" header may instead be placed in a "CC" header going back.

In any reply-all, addresses in a CC header seem to be placed in a CC header in the reply, but not
the address of the person replying if they were in the original CC. They are in the "From" header
going back, so that should be OK, although I don't know what the RFCs say about that, if anything.

Unfortunately, I can't tell you anything about what happens with Bcc headers. I have no messages
containing those.

**Comment by a-name-not-taken** [comment]
Bump

**Comment by kewisch** [comment]
Just acknowledging this issue needs to be looked at, though I don't have an answer on when that might be. Possibly it would be helpful to attach an email that could be used to trigger this behavior.

**Comment by a-name-not-taken** [comment]
Of course, I'm not going to post an unlaundered raw email in a public forum. Maybe to a verifiable developer's email address.

In any case, ANY email that has a "List-Post" header will demonstrate the problem, and that can come from various mailers. The contents of a "List-Post" header should NEVER, repeat NEVER, be used in a reply or reply all message. It just shouldn't be in the code at all. It has no more role than any X- header in steering replies.

For ANY message,, a simple reply should go to the Reply-To address, if present, and, if not, to the From address alone. Never both, and never anywhere else.

**Comment by kewisch** [comment]
Fair enough. I guess I was hoping an eml file could make it easier to reproduce, but then again List-Post is pretty common so this should be easily reproducible without as you mention. And also, it appears this was intentional back then, I found the commit where it started: https://github.com/thunderbird/thunderbird-android/commit/2e33e0dd9492fe2ededf5eaf17495a5dd5ff6f67

I didn't find any RFCs that would support using List-Post in reply, but also I haven't seen any that disallow it. It is likely we need a little more nuance on when to include the List-Post address, or see if there are other ways to fulfill the use cases. I can imagine the original intent was to ensure that responses that are intended for a mailing list go to the mailing list, instead of to the individual you are replying to.

Thanks for getting back quickly!

**Comment by a-name-not-taken** [comment]
Here are the RFCs: It's pretty clear that the ONLY headers relevant to ANY reply are From, To, Reply-To, CC, and Bcc.

https://www.rfc-editor.org/rfc/rfc5322

https://www.rfc-editor.org/rfc/rfc6854

**Comment by a-name-not-taken** [comment]
And, I would add, reply-all would include the mailing list address if the original had been sent to that or CC'd there. If it had been bcc'd to the mailing list, a reply should definitely NOT go there, which it would if the reply were sent to the List-Post.
